src/sentry/notifications/helpers.py

```diff
@@ -610,3 +645,188 @@ def get_providers_for_recipient(
+
+
+def get_all_setting_options(
+    recipient: RpcActor | Team | User | None = None,
+    project: Project | None = None,
+    organization: Organization | None = None,
+    user_ids: Iterable[int] | None = None,
+) -> Iterable[NotificationSettingOption]:
+    """
+    Returns all NotificationSettingOption for given recipients.
+    Recipients can be either a user or a team, or a list of user ids.
+
+    Args:
+        recipient: The recipient of the notification settings (user or team).
+        project: The project to get notification settings for.
+        organization: The organization to get notification settings for.
+        user_ids: The user ids to get notification settings for.
+    """
+    query = get_query(recipient, project, organization, user_ids)
+    return NotificationSettingOption.objects.filter(query)
+
+
+def get_setting_options_for_recipient(
+    recipient: RpcActor | Team | User,
+    project: Project | None = None,
+    organization: Organization | None = None,
+) -> MutableMapping[str, NotificationSettingsOptionEnum]:
+    all_settings = get_all_setting_options(recipient, project, organization)
+
+    notification_settings = {}
+    # Project settings take precedence over all other notification settings
+    for setting in all_settings:
+        if setting.scope_type == NotificationScopeEnum.PROJECT.value:
+            notification_settings[setting.type] = setting.value
+
+    # Organization settings apply when project settings are not set
+    for setting in all_settings:
+        if setting.scope_type == NotificationScopeEnum.ORGANIZATION.value:
+            if setting.type not in notification_settings:
+                notification_settings[setting.type] = setting.value
+
+    # Team/User settings are the most specific.
+    if recipient_is_user(recipient):
+        scope_type = NotificationScopeEnum.USER.value
+    elif recipient_is_team(recipient):
+        scope_type = NotificationScopeEnum.TEAM.value
+    else:
+        raise Exception("recipient must be either user or team")
+
+    for setting in all_settings:
+        if setting.scope_type == scope_type:
+            if setting.type not in notification_settings:
+                notification_settings[setting.type] = setting.value
+
+    # Fill in any missing settings with the default
+    defaults = get_type_defaults()
+    for type in NotificationSettingEnum:
+        if type not in notification_settings and type in defaults:
+            notification_settings[type] = defaults[type]
+
+    return notification_settings
+
+
+def user_has_any_provider_settings(
+    recipient: RpcActor | Team | User, provider: ExternalProviderEnum
+) -> bool:
+    settings = get_all_setting_providers(recipient)
+    for setting in settings:
+        if not setting.provider == provider.value:
+            continue
+
+        if setting.value in {
+            NotificationSettingsOptionEnum.ALWAYS.value,
+            NotificationSettingsOptionEnum.COMMITTED_ONLY.value,
```

Comment:
@snigdhas for provider settings, there is no committed only or subscribe only. Providers are simply on or off.